areas/cloudarchitect/src/AzureMcp.CloudArchitect/Commands/Design/DesignCommand.cs

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.CommandLine;
+using System.CommandLine.Parsing;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using System.Text.Json;
+using AzureMcp.CloudArchitect.Models;
+using AzureMcp.CloudArchitect.Options;
+using AzureMcp.Core.Commands;
+using AzureMcp.Core.Helpers;
+using AzureMcp.Core.Models;
+using Microsoft.Extensions.Logging;
+
+namespace AzureMcp.CloudArchitect.Commands.Design;
+
+public sealed class DesignCommand(ILogger<DesignCommand> logger) : GlobalCommand<ArchitectureDesignToolOptions>
+{
+    private const string CommandTitle = "Design Azure cloud architectures through guided questions";
+    private readonly ILogger<DesignCommand> _logger = logger;
+
+    private readonly Option<string> _questionOption = CloudArchitectOptionDefinitions.Question;
+    private readonly Option<int> _questionNumberOption = CloudArchitectOptionDefinitions.QuestionNumber;
+    private readonly Option<int> _questionTotalQuestions = CloudArchitectOptionDefinitions.TotalQuestions;
+    private readonly Option<string> _answerOption = CloudArchitectOptionDefinitions.Answer;
+    private readonly Option<bool> _nextQuestionNeededOption = CloudArchitectOptionDefinitions.NextQuestionNeeded;
+    private readonly Option<double> _confidenceScoreOption = CloudArchitectOptionDefinitions.ConfidenceScore;
+
+    private readonly Option<string> _architectureDesignToolState = CloudArchitectOptionDefinitions.State;
+
+    private static readonly string s_designArchitectureText = LoadArchitectureDesignText();
+
+    private static string GetArchitectureDesignText() => s_designArchitectureText;
+
+    public override string Name => "design";
+
+    public override string Description => "Azure architecture design tool that gathers requirements through guided questions and recommends optimal solutions.\n\nKey parameters: question, questionNumber, confidenceScore (0.0-1.0, present architecture when ≥0.7), totalQuestions, answer, nextQuestionNeeded, architectureComponent, architectureTier, state.\n\nProcess:\n1. Ask about user role, business goals (1-2 questions at a time)\n2. Track confidence and update requirements (explicit/implicit/assumed)\n3. When confident enough, present architecture with table format, visual organization, ASCII diagrams\n4. Follow Azure Well-Architected Framework principles\n5. Cover all tiers: infrastructure, platform, application, data, security, operations\n6. Provide actionable advice and high-level overview\n\nState tracks components, requirements by category, and confidence factors. Be conservative with suggestions.";
+
+    public override string Title => CommandTitle;
+
+    public override ToolMetadata Metadata => new()
+    {
+        Destructive = false,
+        ReadOnly = true
+    };
+
+    private static string LoadArchitectureDesignText()
+    {
+        Assembly assembly = typeof(DesignCommand).Assembly;
```

Comment:
We need a try-catch here in case something went wrong when it tries to get the file content, since this method was called outside of `ExecuteAsync` and won't be caught by the try-catch in it.